Support TypeScript NodeNext module resolution

### Is your feature request related to a problem? Please describe.

If you're using Node + ESM + TypeScript, packages that export an ESM build to correct ESM import syntax.

### Describe the solution you'd like

Update all imports and exports to have .js extensions.


### Describe alternatives you've considered

I'm not sure there are alternatives.

### Additional context

If you import this package in a project that is using this combination, you get errors like this:

```
node_modules/min-dash/dist/index.d.ts:2:15 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

export * from './collection';
              ~~~~~~~~~~~~~~
```

Also see https://arethetypeswrong.github.io/?p=min-dash%404.1.1

--- a/package.json
+++ b/package.json
@@ -6,6 +6,14 @@
   "module": "dist/index.esm.js",
   "types": "dist/index.d.ts",
   "type": "module",
+  "exports": {
+    ".": {
+      "import": "./dist/index.esm.js",
+      "require": "./dist/index.cjs",
+      "types": "./dist/index.d.ts"
+    },
+    "./package.json": "./package.json"
+  },
   "files": [
     "dist"
   ],

